sections/dashboard/Stake/InputCards/EscrowInputCard.tsx

```diff
@@ -1,112 +1,115 @@
+		return activeTab === 1 && stakedEscrowedKwentaBalance.gt(0) && !!parseFloat(amount);
+	}, [activeTab, stakedEscrowedKwentaBalance, amount]);
+
+	const handleApprove = useCallback(() => {
+		dispatch(approveKwentaToken('kwenta'));
+	}, [dispatch]);
+
+	const handleStakeEscrow = useCallback(() => {
+		dispatch(stakeEscrow(amountBN));
+	}, [dispatch, amountBN]);
+
+	const handleUnstakeEscrow = useCallback(() => {
+		dispatch(unstakeEscrow(amountBN));
+	}, [dispatch, amountBN]);
+
+	const onMaxClick = useCallback(() => {
 		setAmount(truncateNumbers(maxBalance, DEFAULT_TOKEN_DECIMALS));
 	}, [maxBalance]);
 
-	const { config: stakedEscrowKwentaConfig } = usePrepareContractWrite({
-		...rewardEscrowContract,
-		functionName: 'stakeEscrow',
-		args: [amountBN],
-		overrides: {
-			gasLimit: STAKING_LOW_GAS_LIMIT,
-		},
-		enabled: activeTab === 0 && unstakedEscrowedKwentaBalance.gt(0) && !!parseFloat(amount),
-	});
-
-	const { config: unstakedEscrowKwentaConfig } = usePrepareContractWrite({
-		...rewardEscrowContract,
-		functionName: 'unstakeEscrow',
-		args: [amountBN],
-		overrides: {
-			gasLimit: STAKING_LOW_GAS_LIMIT,
-		},
-		enabled: activeTab === 1 && stakedEscrowedBalance.gt(0) && !!parseFloat(amount),
-	});
-
-	const { writeAsync: kwentaApprove } = useContractWrite(kwentaApproveConfig);
-	const { writeAsync: stakeEscrowKwenta } = useContractWrite(stakedEscrowKwentaConfig);
-	const { writeAsync: unstakeEscrowKwenta } = useContractWrite(unstakedEscrowKwentaConfig);
+	const handleTabChange = useCallback((tabIndex: number) => {
+		setAmount('');
+		setActiveTab(tabIndex);
+	}, []);
 
-	const submitEscrow = useCallback(async () => {
-		if (kwentaTokenApproval) {
-			const approveTxn = await kwentaApprove?.();
-			monitorTransaction({
-				txHash: approveTxn?.hash ?? '',
-			});
+	const isDisabled = useMemo(() => {
+		if (!isKwentaTokenApproved) {
+			return false;
 		} else if (activeTab === 0) {
-			const stakeTxn = await stakeEscrowKwenta?.();
-			monitorTransaction({
-				txHash: stakeTxn?.hash ?? '',
-				onTxConfirmed: () => {
-					setAmount('');
-				},
-			});
+			return stakeEnabled;
+		} else {
+			return unstakeEnabled;
+		}
+	}, [isKwentaTokenApproved, activeTab, stakeEnabled, unstakeEnabled]);
+
+	const submitEscrow = useCallback(async () => {
+		if (!isKwentaTokenApproved) {
+			handleApprove();
+		} else if (stakeEnabled) {
+			handleStakeEscrow();
+		} else if (unstakeEnabled) {
+			handleUnstakeEscrow();
+		}
+	}, [
+		handleApprove,
+		isKwentaTokenApproved,
+		handleStakeEscrow,
+		handleUnstakeEscrow,
+		stakeEnabled,
+		unstakeEnabled,
+	]);
+
+	const balance = useMemo(() => {
+		const b = activeTab === 0 ? unstakedEscrowedKwentaBalance : stakedEscrowedKwentaBalance;
+		return truncateNumbers(b, DEFAULT_CRYPTO_DECIMALS);
+	}, [activeTab, unstakedEscrowedKwentaBalance, stakedEscrowedKwentaBalance]);
+
+	const handleChange = useCallback((_: any, newValue: string) => {
+		if (newValue !== '' && newValue.indexOf('.') === -1) {
```

Comment:
Just wondering why we need this extra logic here but it's not something we use on the other number value inputs?

Reply:
It's the same component of the currency input card (exchange page). This is a quick fix to the following issue.
![image](https://user-images.githubusercontent.com/4819006/204591682-4eedbf7c-a7ff-4ece-b067-acf43540c5ac.png)